lib/widgets/message_list.dart

```diff
@@ -1109,7 +1108,13 @@ class StreamMessageRecipientHeader extends StatelessWidget {
         ]));
 
     return GestureDetector(
-      onTap: () => Navigator.push(context,
+      // When already in a topic narrow, disable tap interaction that would just
+      // push a MessageListPage for the same topic narrow.
+      // TODO(#1039) simplify by removing topic-narrow condition if we remove
+      //   recipient headers in topic narrows
+      onTap: (narrow is TopicNarrow)
+        ? null
+        : () => Navigator.push(context,
         MessageListPage.buildRoute(context: context,
           narrow: TopicNarrow.ofMessage(message))),
```

Comment:
nit: indent function body
```suggestion
        : () => Navigator.push(context,
            MessageListPage.buildRoute(context: context,
              narrow: TopicNarrow.ofMessage(message))),
```